File: xinya-backend/src/main/java/com/xinya/dtx/dto/ProDto.java
package com.xinya.dtx.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
class ProSubmitRequest {
    private String patientId;
    private LocalDate recordDate;
    private List<ProAnswer> answers;
}

@Data
@NoArgsConstructor
@AllArgsConstructor
class ProAnswer {
    private String questionId;
    private String answer;
    private Integer score;
}

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ProSubmitResponse {
    private Boolean success;
    private Integer psychEnergyDelta;
    private String message;
    
    public static ProSubmitResponse success() {
        return ProSubmitResponse.builder()
            .success(true)
            .psychEnergyDelta(10)
            .message("打卡成功！您的希望之树获得了成长能量。")
            .build();
    }
}

File: xinya-client/app/src/main/java/com/xinya/dtx/core/network/dto/ProDto.kt
package com.xinya.dtx.core.network.dto

import com.google.gson.annotations.SerializedName

/**
 * PRO提交请求
 */
data class ProSubmitRequest(
    @SerializedName("patient_id")
    val patientId: String,
    @SerializedName("record_date")
    val recordDate: String,
    val answers: List<ProAnswer>
)

data class ProAnswer(
    @SerializedName("question_id")
    val questionId: String,
    val answer: String,
    val score: Int
)

/**
 * PRO提交响应
 */
data class ProSubmitResponse(
    val success: Boolean,
    @SerializedName("psych_energy_delta")
    val psychEnergyDelta: Int = 0,
    val message: String? = null
)
